src/transport_ofi.h

```diff
@@ -299,18 +300,18 @@ void shmem_transport_put_quiet(void)
         success = fi_cntr_read(shmem_transport_ofi_put_cntrfd);
         fail = fi_cntr_readerr(shmem_transport_ofi_put_cntrfd);
 
-        if (success < shmem_transport_ofi_pending_put_counter && fail == 0) {
+        if (success < OPA_load_int(&shmem_transport_ofi_pending_put_counter) && fail == 0) {
             SPINLOCK_BODY();
         }
         else if (fail) {
             struct fi_cq_err_entry e = {0};
             fi_cq_readerr(shmem_transport_ofi_put_nb_cqfd, (void *)&e, 0);
             OFI_CQ_ERROR(shmem_transport_ofi_put_nb_cqfd, &e);
         }
-    } while (success < shmem_transport_ofi_pending_put_counter);
+    } while (success < OPA_load_int(&shmem_transport_ofi_pending_put_counter));
```

Comment:
I think it's possible for a multithreaded run to have ```success``` be temporarily larger than ```pending_put_counter``` if context switches occur at exactly the wrong time since we increment the pending put counter after issuing the OP.  I suggest that we move incrementing the counter to before issuing the OP to resolve this.

I'm also tempted to suggest that we do a single ```OPA_load``` of the counter into a temporary variable.  Currently the first conditional could evaluate to false and the second to true.